### PR TITLE
feat: export and import interfaces as types when using named imports and ESM Module Syntax

### LINKED ISSUE #1793: Support verbatimModuleSyntax in TypeScript Generator
#### Reason/Context
Currently the typescript generator does not explicitly export/import interfaces as types. This is an issue when the typescript option 'verbatimModuleSyntax' is enabled and can potentially cause some other pitfalls.

#### Description
Add a new flag (or make this the default behaviour when using interfaces, afaik that should not cause any issues) that changes the behaviour of 'src/generators/typescript/TypeScriptDependencyManager.ts' so it imports and exports interfaces as types.

Before:
```typescript
import { PublicUser } from './PublicUser'
interface NewStatePayload {
  members: PublicUser[]
}
export { NewStatePayload }
```
After:
```typescript
import type { PublicUser } from './PublicUser'
interface NewStatePayload {
  members: PublicUser[]
}
export type { NewStatePayload }
```

### PR DESCRIPTION
**Pull request by moritzkalwa**
## Description
When using modelType 'interface', exportType 'named' and moduleSystem 'ESM' export and import interfaces as types to support verbatimModuleSyntax.
This should not be a breaking change, interfaces are types even when not using verbatimModuleSyntax, explicitly marking them as types is optional in that case though.

## Related Issue
fixes #1793 

## Checklist
- [x] The code follows the project's coding standards and is properly linted (`npm run lint`).
- [x] Tests have been added or updated to cover the changes.
- [ ] Documentation has been updated to reflect the changes.
- [x] All tests pass successfully locally.(`npm run test`).

## Additional Notes
We might want to document the fact that one needs to use named exports when using interfaces and verbatimModuleSyntax. Im not sure where this documentation should go though, please point me in the right direction here :D

### CODE REVIEW

File: test/generators/typescript/TypeScriptGenerator.spec.ts [hunk 1]
```diff
@@ -128,8 +133,10 @@ ${content}`;
     };
 
     generator = new TypeScriptGenerator({ modelType: 'interface' });
-    const models = await generator.generate(doc);
-    expect(models).toHaveLength(1);
+    const models = await generator.generateCompleteModels(doc, {
+      exportType: 'named'
+    });
+    expect(models).toHaveLength(2);
```
Comment (jonaslagoni) [inline test/generators/typescript/TypeScriptGenerator.spec.ts:139]: If you want to test the generated complete models, please create a test just for this 🙂 As it's two different renderings and what they test :v:

**Review by jonaslagoni — CHANGES_REQUESTED** [review]

**Comment by moritzkalwa** [comment]
@jonaslagoni Do you think it would be okay to change [this example](https://github.com/asyncapi/modelina/blob/master/examples/typescript-interface/index.ts) to use exportType 'named' and moduleSystem 'ESM' to show these changes?

**Comment by jonaslagoni** [comment]
I would duplicate it and create one that shows it with those settings 🙂 You can follow this: https://github.com/asyncapi/modelina/blob/master/docs/contributing.md#adding-examples

File: examples/typescript-interfaces-as-types/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# TypeScript interface
```
Comment (jonaslagoni) [inline examples/typescript-interfaces-as-types/README.md:1]:
```suggestion
# TypeScript interface as types
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just a small change

**Review by jonaslagoni — APPROVED** [review]

**Comment by mikaellanger** [comment]
This makes enum values unusable when using enumType 'enum' since it also exports enums as types not just interfaces, leading to
`MyEnum' cannot be used as a value because it was exported using 'export type'.`
when doing e.g. MyEnum.AnEnumValue

**Comment by jonaslagoni** [comment]
Thanks for raising the issue @mikaellanger

Reverting commit in master.

Taking a second look, this should be targeting `next` because of the breaking changes in the dependency manager: https://github.com/asyncapi/modelina/pull/1800/files#diff-95c504c5267dbd4e90a8ee687814eabaf191637b94093293b968648836d91d39R21

**Pull request merged**